From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Stephen <stephen@radxa.com>
Date: Thu, 30 Jun 2022 09:45:02 +0800
Subject: [PATCH] arm64: dts: rockchip: rock pi 4: fix kernel panic

This patch can help avoid kernel panic.

Kernel panic log:

  [   22.286079] Unable to handle kernel paging request at virtual address 00000261d1000421
  [   22.286787] Mem abort info:
  [   22.287039]   ESR = 0x96000004
  [   22.287318]   EC = 0x25: DABT (current EL), IL = 32 bits
  [   22.287789]   SET = 0, FnV = 0
  [   22.288065]   EA = 0, S1PTW = 0
  [   22.288347] Data abort info:
  [   22.288608]   ISV = 0, ISS = 0x00000004
  [   22.288979]   CM = 0, WnR = 0
  [   22.289249] [00000261d1000421] address between user and kernel address ranges
  [   22.289908] Internal error: Oops: 96000004 [#1] SMP
  [   22.290349] Modules linked in:
  [   22.290667] CPU: 2 PID: 1 Comm: swapper/0 Not tainted 5.10.66-11-rockchip-geee2b32138fd #rockchip
  [   22.291449] Hardware name: Radxa ROCK Pi 4B (DT)
  [   22.291873] pstate: a0000005 (NzCv daif -PAN -UAO -TCO BTYPE=--)
  [   22.292432] pc : rockchip_pll_clk_rate_to_scale+0x6c/0x80
  [   22.292916] lr : rockchip_pll_clk_rate_to_scale+0x48/0x80
  [   22.293398] sp : ffffffc011c8b870
  [   22.293697] x29: ffffffc011c8b870 x28: ffffff80f77872c8
  [   22.294180] x27: ffffffc0118dab00 x26: 000000006b49d200
  [   22.294661] x25: 0000000000000000 x24: ffffff8004cb1200
  [   22.295142] x23: ffffff8002cef800 x22: 0000000000000011
  [   22.295622] x21: 0000000000000000 x20: ffffff80f7830368
  [   22.296104] x19: 000000006b49d200 x18: 0000000000000000
  [   22.296586] x17: 000000000000000e x16: 0000000000000007
  [   22.297067] x15: 000000000000000a x14: 0000000000045c30
  [   22.297548] x13: 0000000000000006 x12: ffffffffffffffff
  [   22.298028] x11: 0000000000000030 x10: 0101010101010101
  [   22.298510] x9 : ffffffc01055338c x8 : ffffff8004cb1280
  [   22.298991] x7 : 0000000000000000 x6 : 000000000000003f
  [   22.299473] x5 : 0000000000000040 x4 : 0000000000000020
  [   22.299954] x3 : 0000000097ffa722 x2 : 0000000000000001
  [   22.300435] x1 : f9000261d1000401 x0 : 0000000000000000
  [   22.300917] Call trace:
  [   22.301147]  rockchip_pll_clk_rate_to_scale+0x6c/0x80
  [   22.301602]  rockchip_adjust_power_scale+0x3c4/0x578
  [   22.302051]  rockchip_cpufreq_adjust_power_scale+0x3c/0x60
  [   22.302542]  dt_cpufreq_probe+0x2b8/0x38c
  [   22.302907]  platform_drv_probe+0x60/0xb0
  [   22.303269]  really_probe+0x2dc/0x46c
  [   22.303601]  driver_probe_device+0x138/0x148
  [   22.303986]  __device_attach_driver+0xd8/0x114
  [   22.304388]  bus_for_each_drv+0xa4/0xcc
  [   22.304735]  __device_attach+0xdc/0x170
  [   22.305082]  device_initial_probe+0x20/0x2c
  [   22.305459]  bus_probe_device+0x3c/0xa0
  [   22.305806]  device_add+0x544/0x6e4
  [   22.306124]  platform_device_add+0xf4/0x22c
  [   22.306501]  platform_device_register_full+0xd8/0x128
  [   22.306958]  platform_device_register_data.constprop.0+0x58/0x80
  [   22.307494]  rockchip_cpufreq_driver_init+0x478/0x4c8
  [   22.307949]  do_one_initcall+0xa0/0x1e8
  [   22.308297]  kernel_init_freeable+0x2a4/0x2ac
  [   22.308690]  kernel_init+0x20/0x11c
  [   22.309008]  ret_from_fork+0x10/0x30
  ...
  [   22.575155] Kernel Offset: disabled
  [   22.575475] CPU features: 0x000,00040022,6100200c
  [   22.575894] Memory Limit: none
  [   22.576184] THERMAL REGS:
  [   22.576433] 00000000: 00000200 00030133 00000031 00000002 00000000 00000000 00000000 00000000
  [   22.577196] 00000020: 0000022e 0000022f 00000000 00000000 0000024e 00000000 00000000 00000000
  [   22.577959] 00000040: 00000279 00000279 00000000 00000000 00000000 00000000 00000000 00000000
  [   22.578722] 00000060: 00000004 00000004 00000753 00000753 00000000 00000000 00000000 00000000
  [   22.579476] 00000080: 00000000 00000000
  [   22.579835] ---[ end Kernel panic - not syncing: Attempted to kill init! exitcode=0x0000000b ]---

Signed-off-by: Stephen <stephen@radxa.com>
Signed-off-by: Radxa <dev@radxa.com>
---
 arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4.dtsi | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4.dtsi b/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4.dtsi
index b60796f2f6a5..79694db04fde 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4.dtsi
@@ -9,6 +9,7 @@
 #include <dt-bindings/pwm/pwm.h>
 #include "rk3399.dtsi"
 #include "rk3399-opp.dtsi"
+#include "rk3399-vop-clk-set.dtsi"
 
 / {
 	clkin_gmac: external-gmac-clock {
-- 
2.25.1

